packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -539,8 +541,9 @@ export class OdspDocumentStorageService implements IDocumentStorageService {
             const odspSnapshot = await this.fetchSnapshotCore(snapshotOptions, tokenFetchOptions, usePost, abortController);
             return odspSnapshot;
         } catch (error) {
-            if (error.errorType === OdspErrorType.snapshotTooBig || error.errorType === OdspErrorType.fetchTimeout) {
-                const snapshotOptionsWithoutBlobs: ISnapshotOptions = { ...snapshotOptions, blobs: undefined };
+            // If the first snapshot request was with blobs and we either timed out or the size was too big, then try to fetch without blobs.
+            if ((error.errorType === OdspErrorType.snapshotTooBig || error.errorType === OdspErrorType.fetchTimeout) && snapshotOptions.blobs) {
```

Comment:
if it's OdspErrorType.snapshotTooBig, but user provided their own mds setting (that is smaller than our default one), then we should throw that error to user.

Reply:
This need to be fixed - that would break Teams - they limit it at 1Mb or so and do not want to load file if it's bigger.

Reply:
Added the check for the same.